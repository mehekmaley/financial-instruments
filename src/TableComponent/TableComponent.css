.Base-tableComponent {
  width: 100%;
  border-collapse: collapse;
}

.Base-tableComponent th, .Base-tableComponent td {
  border: 1px solid #ddd;
  padding: 8px;
  text-align: left;
}

.Base-tableComponent th {
  background-color: #f2f2f2;
  cursor: pointer;
  text-decoration: underline;
}
