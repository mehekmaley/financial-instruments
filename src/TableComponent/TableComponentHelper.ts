export const orderCompare = (a: string, b: string, order: Record<string, number>): number => {
    const aOrder = order[a] || 0; 
    const bOrder = order[b] || 0;
    return aOrder - bOrder;
};

export const getColor = (assetClass: string) => {
    switch (assetClass) {
      case 'Macro':
        return 'white';
      case 'Equities':
        return '#7CB9E8';
      case 'Credit':
        return '#32de84';
      default:
        return 'transparent';
    }
};

export const getPriceColor = (price: number) => (price >= 0 ? 'black' : 'red');
