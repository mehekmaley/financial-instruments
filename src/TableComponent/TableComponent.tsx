import React, { useState, useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import sampleData from '../data/sampleData.json'; 
import './TableComponent.css'; 
import { getColor, getPriceColor, orderCompare } from './TableComponentHelper';

interface FinancialInstrument {
  ticker: string;
  price: number;
  assetClass: string;
}

const TableComponent: React.FC = () => {
    const [data, setData] = useState<FinancialInstrument[]>(sampleData);
    const { t } = useTranslation();

    useEffect(() => {
        sortByAssetClass();
    }, []);

    const sortByAssetClass = () => {
      const sortedData = [...data].sort((a, b) =>
        {
            const order = { Equities: 1, Macro: 2, Credit: 3 };
            return orderCompare(a.assetClass, b.assetClass, order);
        }
      );
      setData(sortedData);
    };
  
    const sortByPrice = () => {
      const sortedData = [...data].sort((a, b) => b.price - a.price);
      setData(sortedData);
    };
  
    const sortByTicker = () => {
      const sortedData = [...data].sort((a, b) =>
        a.ticker.localeCompare(b.ticker)
      );
      setData(sortedData);
    };
    
    const displayScreen = () => {
        return (
            <div>
                <table className='Base-tableComponent'>
                    <thead>
                        <tr>
                            <th className='Base-tableHeader' onClick={sortByAssetClass}>{t('tableHeaders.assetClass')}</th>
                            <th className='Base-tableHeader' onClick={sortByPrice}>{t('tableHeaders.price')}</th>
                            <th className='Base-tableHeader' onClick={sortByTicker}>{t('tableHeaders.ticker')}</th>
                        </tr>
                    </thead>
                    <tbody>
                    {data.map((instrument) => (
                        <tr key={instrument.ticker+'_'+`${index}`} style={{ background: getColor(instrument.assetClass) }}>
                            <td>{instrument.assetClass}</td>
                            <td style={{ color: getPriceColor(instrument.price)}}>{instrument.price}</td>
                            <td>{instrument.ticker}</td>
                        </tr>
                    ))}
                    </tbody>
                </table>
            </div>
        )
    }
  
    return (
        data.length === 0 ? <>{t('loading')}...</> : displayScreen()
    );
  };
  
  export default TableComponent;
  
