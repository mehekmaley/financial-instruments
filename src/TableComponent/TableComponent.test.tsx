import React from 'react';
// import { mount } from 'enzyme';
// import toJson from 'enzyme-to-json';
import { fireEvent, getByTestId, render } from '@testing-library/react';
import TableComponent from './TableComponent';

jest.mock('react-i18next', () => {
  const appEn = {
    financialInstrumentsTable: 'Financial Instruments Table',
    loading: 'Loading',
    tableHeaders: {
        assetClass: 'Asset Class',
        price: 'Price',
        ticker: 'Ticker',
    }
  }
  return {
    useTranslation: () => {
      return {
        t: (str: string) => str.split('.').reduce((o:any, i) => o[i], appEn)
      }
    }
  }
});

describe('TableComponent', () => {
  test('renders loading message when data is empty', () => {
    const wrapper = render(<TableComponent />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders correctly with mock data', () => {
    const mockData = [
      { ticker: 'ALPHA', price: 150.0, assetClass: 'Equities' },
      { ticker: 'BETA', price: 2500.0, assetClass: 'Macro' },
      { ticker: 'GAMMA', price: 300.0, assetClass: 'Credit' },
    ];

    jest.spyOn(React, 'useState').mockReturnValueOnce([mockData, jest.fn()]);

    const wrapper = render(<TableComponent />);
    expect(wrapper).toMatchSnapshot();
  });
}); 