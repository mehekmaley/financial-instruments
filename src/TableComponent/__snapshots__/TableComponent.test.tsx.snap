// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableComponent renders correctly with mock data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table
          class="Base-tableComponent"
        >
          <thead>
            <tr>
              <th
                class="Base-tableHeader"
              >
                Asset Class
              </th>
              <th
                class="Base-tableHeader"
              >
                Price
              </th>
              <th
                class="Base-tableHeader"
              >
                Ticker
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              style="background: rgb(124, 185, 232);"
            >
              <td>
                Equities
              </td>
              <td
                style="color: red;"
              >
                -3791.37
              </td>
              <td>
                BETA
              </td>
            </tr>
            <tr
              style="background: rgb(124, 185, 232);"
            >
              <td>
                Equities
              </td>
              <td
                style="color: black;"
              >
                2299.1
              </td>
              <td>
                GAMMA
              </td>
            </tr>
            <tr
              style="background: rgb(124, 185, 232);"
            >
              <td>
                Equities
              </td>
              <td
                style="color: black;"
              >
                3132.66
              </td>
              <td>
                DELTA
              </td>
            </tr>
            <tr
              style="background: rgb(124, 185, 232);"
            >
              <td>
                Equities
              </td>
              <td
                style="color: black;"
              >
                2735.1
              </td>
              <td>
                OMIKRON
              </td>
            </tr>
            <tr
              style="background: rgb(124, 185, 232);"
            >
              <td>
                Equities
              </td>
              <td
                style="color: black;"
              >
                1854.19
              </td>
              <td>
                SIGMA
              </td>
            </tr>
            <tr
              style="background: rgb(124, 185, 232);"
            >
              <td>
                Equities
              </td>
              <td
                style="color: black;"
              >
                2306.35
              </td>
              <td>
                OMEGA
              </td>
            </tr>
            <tr
              style="background: white;"
            >
              <td>
                Macro
              </td>
              <td
                style="color: black;"
              >
                3089.2
              </td>
              <td>
                ETA
              </td>
            </tr>
            <tr
              style="background: white;"
            >
              <td>
                Macro
              </td>
              <td
                style="color: black;"
              >
                1075.44
              </td>
              <td>
                THETA
              </td>
            </tr>
            <tr
              style="background: white;"
            >
              <td>
                Macro
              </td>
              <td
                style="color: black;"
              >
                1096.64
              </td>
              <td>
                IOTA
              </td>
            </tr>
            <tr
              style="background: white;"
            >
              <td>
                Macro
              </td>
              <td
                style="color: black;"
              >
                2136.64
              </td>
              <td>
                MU
              </td>
            </tr>
            <tr
              style="background: white;"
            >
              <td>
                Macro
              </td>
              <td
                style="color: black;"
              >
                2744.89
              </td>
              <td>
                NU
              </td>
            </tr>
            <tr
              style="background: white;"
            >
              <td>
                Macro
              </td>
              <td
                style="color: black;"
              >
                2082.71
              </td>
              <td>
                TAU
              </td>
            </tr>
            <tr
              style="background: rgb(50, 222, 132);"
            >
              <td>
                Credit
              </td>
              <td
                style="color: red;"
              >
                -3150.67
              </td>
              <td>
                ALPHA
              </td>
            </tr>
            <tr
              style="background: rgb(50, 222, 132);"
            >
              <td>
                Credit
              </td>
              <td
                style="color: black;"
              >
                1168.46
              </td>
              <td>
                EPSILON
              </td>
            </tr>
            <tr
              style="background: rgb(50, 222, 132);"
            >
              <td>
                Credit
              </td>
              <td
                style="color: black;"
              >
                2716.78
              </td>
              <td>
                ZETA
              </td>
            </tr>
            <tr
              style="background: rgb(50, 222, 132);"
            >
              <td>
                Credit
              </td>
              <td
                style="color: black;"
              >
                2321.17
              </td>
              <td>
                KAPPA
              </td>
            </tr>
            <tr
              style="background: rgb(50, 222, 132);"
            >
              <td>
                Credit
              </td>
              <td
                style="color: black;"
              >
                1472.2
              </td>
              <td>
                LAMBDA
              </td>
            </tr>
            <tr
              style="background: rgb(50, 222, 132);"
            >
              <td>
                Credit
              </td>
              <td
                style="color: black;"
              >
                2997.78
              </td>
              <td>
                PSI
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <table
        class="Base-tableComponent"
      >
        <thead>
          <tr>
            <th
              class="Base-tableHeader"
            >
              Asset Class
            </th>
            <th
              class="Base-tableHeader"
            >
              Price
            </th>
            <th
              class="Base-tableHeader"
            >
              Ticker
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            style="background: rgb(124, 185, 232);"
          >
            <td>
              Equities
            </td>
            <td
              style="color: red;"
            >
              -3791.37
            </td>
            <td>
              BETA
            </td>
          </tr>
          <tr
            style="background: rgb(124, 185, 232);"
          >
            <td>
              Equities
            </td>
            <td
              style="color: black;"
            >
              2299.1
            </td>
            <td>
              GAMMA
            </td>
          </tr>
          <tr
            style="background: rgb(124, 185, 232);"
          >
            <td>
              Equities
            </td>
            <td
              style="color: black;"
            >
              3132.66
            </td>
            <td>
              DELTA
            </td>
          </tr>
          <tr
            style="background: rgb(124, 185, 232);"
          >
            <td>
              Equities
            </td>
            <td
              style="color: black;"
            >
              2735.1
            </td>
            <td>
              OMIKRON
            </td>
          </tr>
          <tr
            style="background: rgb(124, 185, 232);"
          >
            <td>
              Equities
            </td>
            <td
              style="color: black;"
            >
              1854.19
            </td>
            <td>
              SIGMA
            </td>
          </tr>
          <tr
            style="background: rgb(124, 185, 232);"
          >
            <td>
              Equities
            </td>
            <td
              style="color: black;"
            >
              2306.35
            </td>
            <td>
              OMEGA
            </td>
          </tr>
          <tr
            style="background: white;"
          >
            <td>
              Macro
            </td>
            <td
              style="color: black;"
            >
              3089.2
            </td>
            <td>
              ETA
            </td>
          </tr>
          <tr
            style="background: white;"
          >
            <td>
              Macro
            </td>
            <td
              style="color: black;"
            >
              1075.44
            </td>
            <td>
              THETA
            </td>
          </tr>
          <tr
            style="background: white;"
          >
            <td>
              Macro
            </td>
            <td
              style="color: black;"
            >
              1096.64
            </td>
            <td>
              IOTA
            </td>
          </tr>
          <tr
            style="background: white;"
          >
            <td>
              Macro
            </td>
            <td
              style="color: black;"
            >
              2136.64
            </td>
            <td>
              MU
            </td>
          </tr>
          <tr
            style="background: white;"
          >
            <td>
              Macro
            </td>
            <td
              style="color: black;"
            >
              2744.89
            </td>
            <td>
              NU
            </td>
          </tr>
          <tr
            style="background: white;"
          >
            <td>
              Macro
            </td>
            <td
              style="color: black;"
            >
              2082.71
            </td>
            <td>
              TAU
            </td>
          </tr>
          <tr
            style="background: rgb(50, 222, 132);"
          >
            <td>
              Credit
            </td>
            <td
              style="color: red;"
            >
              -3150.67
            </td>
            <td>
              ALPHA
            </td>
          </tr>
          <tr
            style="background: rgb(50, 222, 132);"
          >
            <td>
              Credit
            </td>
            <td
              style="color: black;"
            >
              1168.46
            </td>
            <td>
              EPSILON
            </td>
          </tr>
          <tr
            style="background: rgb(50, 222, 132);"
          >
            <td>
              Credit
            </td>
            <td
              style="color: black;"
            >
              2716.78
            </td>
            <td>
              ZETA
            </td>
          </tr>
          <tr
            style="background: rgb(50, 222, 132);"
          >
            <td>
              Credit
            </td>
            <td
              style="color: black;"
            >
              2321.17
            </td>
            <td>
              KAPPA
            </td>
          </tr>
          <tr
            style="background: rgb(50, 222, 132);"
          >
            <td>
              Credit
            </td>
            <td
              style="color: black;"
            >
              1472.2
            </td>
            <td>
              LAMBDA
            </td>
          </tr>
          <tr
            style="background: rgb(50, 222, 132);"
          >
            <td>
              Credit
            </td>
            <td
              style="color: black;"
            >
              2997.78
            </td>
            <td>
              PSI
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TableComponent renders loading message when data is empty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table
          class="Base-tableComponent"
        >
          <thead>
            <tr>
              <th
                class="Base-tableHeader"
              >
                Asset Class
              </th>
              <th
                class="Base-tableHeader"
              >
                Price
              </th>
              <th
                class="Base-tableHeader"
              >
                Ticker
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              style="background: rgb(124, 185, 232);"
            >
              <td>
                Equities
              </td>
              <td
                style="color: red;"
              >
                -3791.37
              </td>
              <td>
                BETA
              </td>
            </tr>
            <tr
              style="background: rgb(124, 185, 232);"
            >
              <td>
                Equities
              </td>
              <td
                style="color: black;"
              >
                2299.1
              </td>
              <td>
                GAMMA
              </td>
            </tr>
            <tr
              style="background: rgb(124, 185, 232);"
            >
              <td>
                Equities
              </td>
              <td
                style="color: black;"
              >
                3132.66
              </td>
              <td>
                DELTA
              </td>
            </tr>
            <tr
              style="background: rgb(124, 185, 232);"
            >
              <td>
                Equities
              </td>
              <td
                style="color: black;"
              >
                2735.1
              </td>
              <td>
                OMIKRON
              </td>
            </tr>
            <tr
              style="background: rgb(124, 185, 232);"
            >
              <td>
                Equities
              </td>
              <td
                style="color: black;"
              >
                1854.19
              </td>
              <td>
                SIGMA
              </td>
            </tr>
            <tr
              style="background: rgb(124, 185, 232);"
            >
              <td>
                Equities
              </td>
              <td
                style="color: black;"
              >
                2306.35
              </td>
              <td>
                OMEGA
              </td>
            </tr>
            <tr
              style="background: white;"
            >
              <td>
                Macro
              </td>
              <td
                style="color: black;"
              >
                3089.2
              </td>
              <td>
                ETA
              </td>
            </tr>
            <tr
              style="background: white;"
            >
              <td>
                Macro
              </td>
              <td
                style="color: black;"
              >
                1075.44
              </td>
              <td>
                THETA
              </td>
            </tr>
            <tr
              style="background: white;"
            >
              <td>
                Macro
              </td>
              <td
                style="color: black;"
              >
                1096.64
              </td>
              <td>
                IOTA
              </td>
            </tr>
            <tr
              style="background: white;"
            >
              <td>
                Macro
              </td>
              <td
                style="color: black;"
              >
                2136.64
              </td>
              <td>
                MU
              </td>
            </tr>
            <tr
              style="background: white;"
            >
              <td>
                Macro
              </td>
              <td
                style="color: black;"
              >
                2744.89
              </td>
              <td>
                NU
              </td>
            </tr>
            <tr
              style="background: white;"
            >
              <td>
                Macro
              </td>
              <td
                style="color: black;"
              >
                2082.71
              </td>
              <td>
                TAU
              </td>
            </tr>
            <tr
              style="background: rgb(50, 222, 132);"
            >
              <td>
                Credit
              </td>
              <td
                style="color: red;"
              >
                -3150.67
              </td>
              <td>
                ALPHA
              </td>
            </tr>
            <tr
              style="background: rgb(50, 222, 132);"
            >
              <td>
                Credit
              </td>
              <td
                style="color: black;"
              >
                1168.46
              </td>
              <td>
                EPSILON
              </td>
            </tr>
            <tr
              style="background: rgb(50, 222, 132);"
            >
              <td>
                Credit
              </td>
              <td
                style="color: black;"
              >
                2716.78
              </td>
              <td>
                ZETA
              </td>
            </tr>
            <tr
              style="background: rgb(50, 222, 132);"
            >
              <td>
                Credit
              </td>
              <td
                style="color: black;"
              >
                2321.17
              </td>
              <td>
                KAPPA
              </td>
            </tr>
            <tr
              style="background: rgb(50, 222, 132);"
            >
              <td>
                Credit
              </td>
              <td
                style="color: black;"
              >
                1472.2
              </td>
              <td>
                LAMBDA
              </td>
            </tr>
            <tr
              style="background: rgb(50, 222, 132);"
            >
              <td>
                Credit
              </td>
              <td
                style="color: black;"
              >
                2997.78
              </td>
              <td>
                PSI
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <table
        class="Base-tableComponent"
      >
        <thead>
          <tr>
            <th
              class="Base-tableHeader"
            >
              Asset Class
            </th>
            <th
              class="Base-tableHeader"
            >
              Price
            </th>
            <th
              class="Base-tableHeader"
            >
              Ticker
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            style="background: rgb(124, 185, 232);"
          >
            <td>
              Equities
            </td>
            <td
              style="color: red;"
            >
              -3791.37
            </td>
            <td>
              BETA
            </td>
          </tr>
          <tr
            style="background: rgb(124, 185, 232);"
          >
            <td>
              Equities
            </td>
            <td
              style="color: black;"
            >
              2299.1
            </td>
            <td>
              GAMMA
            </td>
          </tr>
          <tr
            style="background: rgb(124, 185, 232);"
          >
            <td>
              Equities
            </td>
            <td
              style="color: black;"
            >
              3132.66
            </td>
            <td>
              DELTA
            </td>
          </tr>
          <tr
            style="background: rgb(124, 185, 232);"
          >
            <td>
              Equities
            </td>
            <td
              style="color: black;"
            >
              2735.1
            </td>
            <td>
              OMIKRON
            </td>
          </tr>
          <tr
            style="background: rgb(124, 185, 232);"
          >
            <td>
              Equities
            </td>
            <td
              style="color: black;"
            >
              1854.19
            </td>
            <td>
              SIGMA
            </td>
          </tr>
          <tr
            style="background: rgb(124, 185, 232);"
          >
            <td>
              Equities
            </td>
            <td
              style="color: black;"
            >
              2306.35
            </td>
            <td>
              OMEGA
            </td>
          </tr>
          <tr
            style="background: white;"
          >
            <td>
              Macro
            </td>
            <td
              style="color: black;"
            >
              3089.2
            </td>
            <td>
              ETA
            </td>
          </tr>
          <tr
            style="background: white;"
          >
            <td>
              Macro
            </td>
            <td
              style="color: black;"
            >
              1075.44
            </td>
            <td>
              THETA
            </td>
          </tr>
          <tr
            style="background: white;"
          >
            <td>
              Macro
            </td>
            <td
              style="color: black;"
            >
              1096.64
            </td>
            <td>
              IOTA
            </td>
          </tr>
          <tr
            style="background: white;"
          >
            <td>
              Macro
            </td>
            <td
              style="color: black;"
            >
              2136.64
            </td>
            <td>
              MU
            </td>
          </tr>
          <tr
            style="background: white;"
          >
            <td>
              Macro
            </td>
            <td
              style="color: black;"
            >
              2744.89
            </td>
            <td>
              NU
            </td>
          </tr>
          <tr
            style="background: white;"
          >
            <td>
              Macro
            </td>
            <td
              style="color: black;"
            >
              2082.71
            </td>
            <td>
              TAU
            </td>
          </tr>
          <tr
            style="background: rgb(50, 222, 132);"
          >
            <td>
              Credit
            </td>
            <td
              style="color: red;"
            >
              -3150.67
            </td>
            <td>
              ALPHA
            </td>
          </tr>
          <tr
            style="background: rgb(50, 222, 132);"
          >
            <td>
              Credit
            </td>
            <td
              style="color: black;"
            >
              1168.46
            </td>
            <td>
              EPSILON
            </td>
          </tr>
          <tr
            style="background: rgb(50, 222, 132);"
          >
            <td>
              Credit
            </td>
            <td
              style="color: black;"
            >
              2716.78
            </td>
            <td>
              ZETA
            </td>
          </tr>
          <tr
            style="background: rgb(50, 222, 132);"
          >
            <td>
              Credit
            </td>
            <td
              style="color: black;"
            >
              2321.17
            </td>
            <td>
              KAPPA
            </td>
          </tr>
          <tr
            style="background: rgb(50, 222, 132);"
          >
            <td>
              Credit
            </td>
            <td
              style="color: black;"
            >
              1472.2
            </td>
            <td>
              LAMBDA
            </td>
          </tr>
          <tr
            style="background: rgb(50, 222, 132);"
          >
            <td>
              Credit
            </td>
            <td
              style="color: black;"
            >
              2997.78
            </td>
            <td>
              PSI
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
